django/applications/catmaid/static/js/widgets/3dviewer.js

```diff
@@ -23,6 +23,12 @@
    * compute with web workers (betweenness centrality shading). */
   var WebGLApplication = function() {
     this.widgetID = this.registerInstance();
+
+    if (CATMAID.MAX_WEBGL_CONTEXTS - CATMAID.countWebGlContexts() < 1) {
+      CATMAID.TooManyWebGlContextsDialog().show();
```

Comment:
I think an exception would be the way to go. As an example, a few are defined in `static/libs/catmaid/error.js`. This of course would require the Window Maker to try/catch window creation, which it should do anyway. For catching it could delegate to `CATMAID.handleError()`.